python/tvm/relay/testing/tflite.py

```diff
@@ -0,0 +1,468 @@
+            The scale(s)
+        zero_point: List[int]
+            The zero point(s)
+        quantized_dimension: int
+            The dimension across which quantization is applied
+        """
+        self.scale = scale
+        self.zero_point = zero_point
+        self.quantized_dimension = quantized_dimension
+
+    def to_json(self) -> Dict[str, Any]:
+        return {
+            "scale": self.scale,
+            "zero_point": self.zero_point,
+            "quantized_dimension": self.quantized_dimension,
+        }
+
+
+class Tensor:
+    """A class representing a tensor"""
+
+    def __init__(
+        self,
+        data_type: str,
+        shape: List[int],
+        quantization: Optional[Quantization] = None,
+        buffer_data: Optional[List[int]] = None,
+    ):
+        """
+        Parameters
+        ----------
+        data_type: str
+            The data type of data in the tensor
+        shape: List[int]
+            The shape of the tensor
+        quantization: Optional[Quantization]
+            The quantization parameters of the tensor
+        buffer_data: Optional[List[int]]
+            The data in the tensor
+        """
+        self.data_type = data_type
+        self.buffer_idx = None
+        self.name = None
+        self.shape = shape
+        self.quantization = quantization
+        self.buffer_data = buffer_data
+
+    def to_json(self) -> Dict[str, Any]:
+        tensor_json = {
+            "type": self.data_type.upper(),
+            "buffer": self.buffer_idx,
+            "name": self.name,
+            "shape": self.shape,
+        }
+        if self.quantization is not None:
+            tensor_json["quantization"] = self.quantization.to_json()
+        return tensor_json
+
+
+class Operator:
+    """A class representing an operator"""
+
+    def __init__(
+        self,
+        opcode: int,
+        options_type: str,
+        options: Dict[str, Any],
+    ):
+        """
+        Parameters
+        ----------
+        opcode: int
+            The operator's builtin_code
+        options_type: str
+            The operator's builtin_options_type
+        options: Dict[str, Any]
+            The operator's builtin_options
+        """
+        self.opcode = opcode
+        self.options_type = options_type
+        self.options = options
+        self.op_inputs_idx = []
+        self.op_outputs_idx = []
+
+
+def generate_tflite_model(
+    inputs: List[Tensor],
+    outputs: List[Tensor],
+    operator: Operator,
+) -> bytes:
+    """Generate a TensorFlow Lite model
+
+    Parameters
+    ----------
+    inputs: List[Tensor],
+        The list of input tensors
+    outputs: List[Tensor],
+        The list of output tensors
+    operator: Operator,
+        The operator in the model
+
+    Returns
+    ------------
+    TensorFlow Lite model as bytes
+    """
+    tmp_dir = tempfile.gettempdir()
+
+    schema_path = tmp_dir + "/schema.fbs"
+
+    download(SCHEMA_URL, schema_path)
```

Comment:
Just to confirm, does this cache the schema between runs of this function or will it try and download it every time?

Reply:
I will cache it! I'll add a comment in the code